src/gbm/gblinear_model.h

```diff
@@ -10,6 +10,8 @@
 #include <string>
 #include <cstring>
 
+#include "xgboost/json.h"
```

Comment:
I would suggest to put a forward declaration here. `class Json;` should be enough
Otherwise `json.h` header could "escape" though this header.